### ISSUE TITLE
Fix Request: false positive and missed case in UnconditionalCreatedLogArguments

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#209]
Example:
```
class Foo {
    Bar field1;
    Bar field2;

    void bad(Bar param) {
      String log = param.baz(field1.get());
      String logThis = param.baz(this.field1.get());
      log.debug("field: {}", log); //bad, missed
      log.debug("field: {}", logThis); //bad

      String fieldGet = field1.getClass();
      String fieldGetThis = this.field2.getClass();
      log.debug("field getter: {}", fieldGet); // good
      log.debug("field this.getter: {}", fieldGetThis); // good, false positive 
    }
}
```
